### PR TITLE
Add retry mechanism to get_parquet_file in parquet metadata step

### PR DESCRIPTION
**Pull request by polinaeterna**
...to see if it helps with [FineWeb](https://huggingface.co/datasets/HuggingFaceFW/fineweb) `config-parquet-metadata` issue. 

Currently the error says just `Server disconnected` which seems to be a `aiohttp.ServerDisconnectedError` error.

If that works, a more fundamental solution would be to completely switch to `HfFyleSystem` instead of `HTTPFileSystem` and remove retries.

### CODE REVIEW

File: services/worker/src/worker/job_runners/config/parquet_metadata.py [hunk 1]
```diff
@@ -31,7 +33,10 @@ def create_parquet_metadata_file_from_remote_parquet(
     parquet_file_item: SplitHubFile, fs: HTTPFileSystem, hf_token: Optional[str], parquet_metadata_directory: StrPath
 ) -> ParquetFileMetadataItem:
     try:
-        parquet_file = get_parquet_file(url=parquet_file_item["url"], fs=fs, hf_token=hf_token)
+        retry_get_parquet_file = retry(on=[aiohttp.ServerConnectionError], sleeps=HF_HUB_HTTP_ERROR_RETRY_SLEEPS)(
```
Comment (polinaeterna) [inline services/worker/src/worker/job_runners/config/parquet_metadata.py:36]: or should it be specifically `ServerDisconnectedError`?

Comment (lhoestq) [reply services/worker/src/worker/job_runners/config/parquet_metadata.py:36]: if ServerConnectionError is a parent class of ServerDisconnectedError it's ok imo

Comment (lhoestq) [reply services/worker/src/worker/job_runners/config/parquet_metadata.py:36]: btw `retry` will do a `sleep` before each first call to `get_parquet_file`, which can make the parquet-metadata job a little bit slower (it's unlikely to be longer that the maximum allowed 40min but better keep this in mind when re-running it for datasets like fineweb)

**Review by lhoestq — APPROVED** [review]
lgtm !

File: services/worker/src/worker/job_runners/config/parquet_metadata.py (hunk 1 above)
Comment (polinaeterna) [reply services/worker/src/worker/job_runners/config/parquet_metadata.py]: I changed the first value of sleeps from 1 to 0.2

**Comment by polinaeterna** [comment]
seems that it helped! but it was slow indeed :( almost half an hour
image
image

**Comment by lhoestq** [comment]
Great ! Given the number of files it's not absurd ^^

**Pull request merged**